src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/ReviewVisibleEventHandlers.cs

```diff
@@ -39,17 +41,43 @@ public abstract class ReviewVisibleEventHandlersAnalyzer : DiagnosticAnalyzer
                                                                              isDataflowRule: false,
                                                                              isEnabledByDefaultInFxCopAnalyzers: false);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray<DiagnosticDescriptor>.Empty;
-        //ImmutableArray.Create(SecurityRule, DefaultRule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(SecurityRule, DefaultRule);
 
-#pragma warning disable RS1025 // Configure generated code analysis
         public override void Initialize(AnalysisContext analysisContext)
-#pragma warning restore RS1025 // Configure generated code analysis
         {
             analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
-            // TODO: Configure generated code analysis.
-            //analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            analysisContext.RegisterCompilationStartAction(context =>
+            {
+                var eventArgsType = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemEventArgs);
+                var securityPermissionAttributeType = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemSecurityPermissionsSecurityPermissionAttribute);
+
+                context.RegisterSymbolAction(context =>
+                {
+                    var method = (IMethodSymbol)context.Symbol;
+
+                    // FxCop compat: only analyze externally visible symbols by default.
+                    if (!method.MatchesConfiguredVisibility(context.Options, DefaultRule, context.CancellationToken))
+                    {
+                        return;
+                    }
+
+                    if (method.IsOverride || method.IsImplementationOfAnyInterfaceMember())
+                    {
+                        return;
+                    }
+
+                    if (method.HasEventHandlerSignature(eventArgsType))
+                    {
+                        var rule = method.HasAttribute(securityPermissionAttributeType)
```

Comment:
I don't know if there is any other criteria triggering the "security" part of the rule.

Reply:
Seems the security rule fired only for cases which are now deprecated:

```csharp
    void DoSecurityCheck(Method method)
    {      
      ParameterCollection parameters = method.Parameters;

      for (int i = 0; i < parameters.Count; i++)
      {
        Parameter parameter = parameters[i];
        if (RuleUtilities.IsEventArgs(parameter.Type)
          && SecurityRuleUtilities.HasLinkDemandOrDemandSecurityAction(method))
```

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.security.permissions.securityaction?view=netframework-4.8

I think we should remove the security rule.

Reply:
Do you want to simply remove the rule or make it not raising for netcore + net fwk >= 4.8?

Reply:
I'd say let just remove the rule as it is pertains to an obsolete API